lib/puppet/provider/exec/powershell.rb

```diff
@@ -68,7 +68,13 @@ def run(command, check = false)
         return super("cmd.exe /c \"\"#{native_path(command(:powershell))}\" #{legacy_args} -Command - < \"#{native_path}\"\"", check)
       end
     else
-      result = ps_manager.execute(command)
+      if working_dir = resource[:cwd]
+        self.fail "Working directory '#{working_dir}' does not exist" unless File.directory?(working_dir)
```

Comment:
This is hit when a resource is executed (onlyif, unless or command).  It's a guard to stop trying to execute Powershell if the working dir doesn't exist.  There is a guard in the powershell manager similar to this.  In practice this particular check isn't required as it will be picked up in the PS manager but I preferred to fail early.

Thinking about it, this will fail the resource in and onlyif/unless situation whereas If I just left it in the PS manager, it will evaluate to a non-zero exit code.  These scenarios are different